### PR TITLE
feat: register the taoPublishing search using proxy

### PR DESCRIPTION
**Pull request by gabrielfs7**
Make taoPublishing use the proxy

https://oat-sa.atlassian.net/browse/ADF-258

Related PRs:

- [x] https://github.com/oat-sa/tao-core/pull/2933
- [x] https://github.com/oat-sa/lib-tao-elasticsearch/pull/45
- [x] https://github.com/oat-sa/extension-tao-advanced-search/pull/27

### CODE REVIEW

File: migrations/Version202106011349443635_taoPublishing.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+{
+    public function getDescription(): string
+    {
+        return 'Registering listener for RemoteDeliveryCreatedEvent';
+    }
+
+    public function up(Schema $schema): void
+    {
+        /** @var SearchProxy $searchProxy */
+        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
```
Comment (gabrielfs7) [inline migrations/Version202106011349443635_taoPublishing.php:42]: Just a remind to not merge before get updated version of tao-core in the composer.json

File: migrations/Version202106011349443635_taoPublishing.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+final class Version202106011349443635_taoPublishing extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Registering ' . TaoPublishingGenerisSearch::class;
+    }
+
+    public function up(Schema $schema): void
+    {
+        $registerGenerisSearch = new RegisterGenerisSearch();
```
Comment (boajer) [inline migrations/Version202106011349443635_taoPublishing.php:40]: I believe you'll need to do something like that to access service locator within your service 
```suggestion
        $registerGenerisSearch = new RegisterGenerisSearch();
        $this->propagate($registerGenerisSearch);
```

File: migrations/Version202106011349443635_taoPublishing.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        $registerGenerisSearch->__invoke([]);
+    }
+
+    public function down(Schema $schema): void
+    {
+        $unRegisterGenerisSearch = new UnRegisterGenerisSearch();
```
Comment (boajer) [inline migrations/Version202106011349443635_taoPublishing.php:46]: Please see https://github.com/oat-sa/extension-tao-publishing/pull/111/files#r643838320

File: scripts/install/RegisterGenerisSearch.php [hunk 4]
```diff
@@ -1,25 +1,53 @@
...
-            $this->getServiceManager()->register(Search::SERVICE_ID, $newSearchService);
+        /** @var SearchProxy $searchProxy */
+        $searchProxy = $this->getServiceManager()->get(SearchProxy::SERVICE_ID);
+
+        /** @var OriginalGenerisSearch $generisSearch */
+        $generisSearch = $searchProxy->getOption(SearchProxy::OPTION_DEFAULT_SEARCH_CLASS);
+
+        if ($generisSearch instanceof OriginalGenerisSearch) {
+            $searchProxy->setOption(
+                SearchProxy::OPTION_DEFAULT_SEARCH_CLASS,
```
Comment (boajer) [inline scripts/install/RegisterGenerisSearch.php:44]: Just a note to update tao-core version on composer before merge
- [ ] Update tao-core

File: migrations/Version202106011349443635_taoPublishing.php (hunk 2 above)
Comment (gabrielfs7) [reply migrations/Version202106011349443635_taoPublishing.php:40]: This file was removed. I do not know why it is here. I will try to recreate and delete again

File: migrations/Version202106011349443635_taoPublishing.php (hunk 3 above)
Comment (gabrielfs7) [reply migrations/Version202106011349443635_taoPublishing.php]: File was removed :)

**Review by boajer — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [X New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Comment by AntonNeviarouski** [comment]
QA verified on http://aadvanced-search-integration.playground.kitchen.it.taocloud.org:48321/

**Pull request merged**